tests/01-deploy.py

```diff
@@ -27,6 +28,45 @@ def test_deploy(self):
         output, retcode = self.unit.run("pgrep -a java")
         assert 'Kafka' in output, "Kafka daemon is not started"
 
+    def test_bind_network_interface(self):
+        '''
+        Test to verify that we can bind to a specific network interface.
+
+        '''
+        self.d.log.debug("Binding Kafka to eth0")
+        self.d.configure('kafka', {'network_interface': 'eth0'})
+        self.d.sentry.wait_for_messages({'kafka': 'Server config changed: restarting Kafka'}, timeout=60)
+        self.d.sentry.wait_for_messages({'kafka': 'Ready'}, timeout=600)
+
+        self.d.log.debug("Checking kafka bindings ...")
+        ret = self.unit.run(
+            'grep host.name /etc/kafka/conf/server.properties')[0]
+        # Correct line should start with host.name (no comment hash
+        # mark), followed by an equals sign and something that looks
+        # like an IP address (we aren't too strict about it being a
+        # valid ip address.)
+        matcher = re.compile("^host\.name=\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}.*")
+
+        self.assertTrue('host.name' in ret)
+        self.assertTrue(matcher.match(ret))
+
+    def test_reset_network_interface(self):
+        """
+        Verify that we can reset the charm to listen on any network interface.
+
+        """
+        self.d.log.debug("Binding Kafka to 0.0.0.0")
+        self.d.configure('kafka', {'network_interface': '0.0.0.0'})
+        self.d.sentry.wait_for_messages({'kafka': 'Server config changed: restarting Kafka'}, timeout=60)
```

Comment:
johnsca: the tricky thing with using status history is that we don't want to catch previous instances of this message, that have been triggered by previous tests. We don't officially know whether this test will execute first, second, or third, for example, so we don't know whether we're looking for one, two or three instances of the string.

Maybe it would make sense to just catch timeout errors here, and attempt the next part of the test; it will fail if the code hasn't been run, anyway.